test/app/model/ChessEngine.spec.ts

```diff
@@ -0,0 +1,145 @@
+    const squareFrom: Square = { column: 'A', row: 2 };
+    const squareTo: Square = { column: 'A', row: 3 };
+
+    expect(engine.move(player, squareFrom, squareTo)).toIncludeSameMembers([
+      {
+        eventType: 'PieceWasMoved',
+        piece: whitePiece,
+        from: squareFrom,
+        to: squareTo,
+      },
+    ]);
+  });
+
+  it('Given white piece on A2 and black piece on A4, when move white piece from A2 to A4, then white piece was moved from A2 to A4 and piece from A4 was captured', () => {
+    const whitePiece = new Queen(Side.WHITE);
+    const blackPiece = new Pawn(Side.BLACK);
+    const boardWithPieces: SquareWithPiece = { A2: whitePiece, A4: blackPiece };
+    const chessBoard = new ChessBoard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const player = new Player(Side.WHITE);
+    const squareFrom: Square = { column: 'A', row: 2 };
+    const squareTo: Square = { column: 'A', row: 4 };
+
+    expect(engine.move(player, squareFrom, squareTo)).toIncludeSameMembers([
+      { eventType: 'PieceWasCaptured', piece: blackPiece, onSquare: squareTo },
+      { eventType: 'PieceWasMoved', piece: whitePiece, from: squareFrom, to: squareTo },
+    ]);
+  });
+
+  it('Given white pieces on C2 and B1 and black piece on B4, when move white piece to C3, then black piece attacks on C3 and then another white piece attacks on C3', () => {
+    const whitePawn = new Pawn(Side.WHITE);
+    const whiteKnight = new Knight(Side.WHITE);
+    const blackPawn = new Pawn(Side.BLACK);
+    const boardWithPieces: SquareWithPiece = { C2: whitePawn, B1: whiteKnight, B4: blackPawn };
+    const chessBoard = new ChessBoard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const playerWhite = new Player(Side.WHITE);
+    const playerBlack = new Player(Side.BLACK);
+
+    engine.move(playerWhite, { column: 'C', row: 2 }, { column: 'C', row: 3 });
+    engine.move(playerBlack, { column: 'B', row: 4 }, { column: 'C', row: 3 });
+    expect(engine.move(playerWhite, { column: 'B', row: 1 }, { column: 'C', row: 3 })).toIncludeSameMembers([
+      { eventType: 'PieceWasCaptured', piece: blackPawn, onSquare: { column: 'C', row: 3 } },
+      { eventType: 'PieceWasMoved', piece: whiteKnight, from: { column: 'B', row: 1 }, to: { column: 'C', row: 3 } },
+    ]);
+  });
+
+  it('Should throw an Error if player wants to move opponents piece', () => {
+    const whitePiece = new Queen(Side.WHITE);
+    const blackPiece = new Pawn(Side.BLACK);
+    const boardWithPieces: SquareWithPiece = { A2: blackPiece, A4: whitePiece };
+    const chessBoard = new ChessBoard(boardWithPieces);
+    const engine = new ChessEngine(chessBoard);
+    const player = new Player(Side.WHITE);
+    const squareFrom: Square = { column: 'A', row: 2 };
+    const squareTo: Square = { column: 'A', row: 4 };
+
+    expect(() => engine.move(player, squareFrom, squareTo)).toThrowError('Player can not move other players pieces.');
```

Comment:
pewnie niezła rozkmina była jak to wywołać, zgadłem :-) ?

Reply:
dokumentacja jest szybko rozwiała wątpliwości :P